### PR TITLE
fix(xmlupload): allow non-prefixed custom groups, and custom groups with dash (DEV-3979)

### PR DESCRIPTION
**Pull request by jnussbaum**

### CODE REVIEW

File: testdata/xml-data/test-data-systematic.xml
Comment (jnussbaum) [inline testdata/xml-data/test-data-systematic.xml:1]: These 2 new lines produced an error before, now they work. I need this for Rita's LIMC upload.

**Review by Nora-Olivia-Ammann — APPROVED** [review]
I don't object to the code, but I am not sure if we even want to create those groups.

**Comment by jnussbaum** [comment]
> I don't object to the code, but I am not sure if we even want to create those groups.

Until recently, I thought we don't need them, but for LIMC, we indeed need it. There, we have the following access levels:
- KnownUsers: view some resources
- ProjectMembers: view all resources
- limc-editors: modify + delete
- ProjectAdmins: admin

This cannot be solved without custom groups, unfortunately.

**Comment by Nora-Olivia-Ammann** [comment]
Does it make sense to create these this way even though we will change it in the future? Is Rita aware of this and okayed it?

**Review by BalduinLandolt — APPROVED** [review]

**Pull request merged**